monai/networks/trt_compile.py

```diff
@@ -0,0 +1,532 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import inspect
+import os
+import tempfile
+import threading
+from collections import OrderedDict
+from pathlib import Path
+from types import MethodType
+
+import torch
+
+from monai.apps.utils import get_logger
+from monai.networks.utils import add_casts_around_norms, convert_to_onnx, convert_to_torchscript
+from monai.utils.module import optional_import
+
+polygraphy, polygraphy_imported = optional_import("polygraphy")
+if polygraphy_imported:
+    from polygraphy.backend.common import bytes_from_path
+    from polygraphy.backend.trt import (
+        CreateConfig,
+        Profile,
+        engine_bytes_from_network,
+        engine_from_bytes,
+        network_from_onnx_path,
+    )
+
+trt, trt_imported = optional_import("tensorrt")
+torch_tensorrt, _ = optional_import("torch_tensorrt", "1.4.0")
+cudart, _ = optional_import("cuda.cudart")
+
+
+lock_sm = threading.Lock()
+
+
+# Map of TRT dtype -> Torch dtype
+def trt_to_torch_dtype_dict():
+    return {
+        trt.int32: torch.int32,
+        trt.float32: torch.float32,
+        trt.float16: torch.float16,
+        trt.bfloat16: torch.float16,
+        trt.int64: torch.int64,
+        trt.int8: torch.int8,
+        trt.bool: torch.bool,
+    }
+
+
+def get_dynamic_axes(profiles):
+    """
+    Given [[min,opt,max],...] list of profile dimensions,
+    this method calculates dynamic_axes to use in onnx.export()
```

Comment:
For  public functions, can we add the docstring for inputs and outputs. Alternatively, we can make this for internal use only `_get_dynamic_axes`. Thanks @borisfom !